### DISCUSSION TITLE
Feature Request: allow custom folder/URL for config file

### DISCUSSION THREAD

**Discussion by txoof** [rxhanson/RectanglePro-Community#88] [Ideas]
It would be great to be able to specify a remote URL or an alternate directory for the config file; this would allow syncing across multiple devices without the need for iCloud. The file can then be synchronized across devices without the need to manually import/export across each device each time a config change is made.

This would be ideal for users that do not, or cannot use iCloud for sync (e.g. managed work computer & home computer). For example, I use a Git repo to synchronize all my dotfiles across multiple devices and keep some sort of change log.

[iTerm2 offers an option to use a custom folder/URL](https://iterm2.com/documentation-one-page.html#documentation-preferences.html)

> ### Preferences
> **Load preferences from a custom folder or URL:**
> If enabled, iTerm2 will load its preferences from the specified folder or URL. After setting this, you'll be prompted when you quit iTerm2 if you'd like to save changes to the folder.

**Comment by rxhanson** [comment]
Thanks for this. I have wanted to add this in, and I was not aware of the way iTerm2 did this. That sounds like the best way.

**Reply by txoof** [reply]
I browsed through their code and it seems pretty well documented though a bit over my head. Maybe there are some ideas you can lift from their good thinking. 

Thanks for considering adding this feature. I recently discovered RectanglePro from The Verge (?) and I really love it. Thanks for creating this!

**Comment by nosleduc** [comment]
Same case here I would like to synchronize configuration between two macs without using iCloud. Is it possible now ?

**Reply by rxhanson** [reply]
Not as good as the ask here, but the upon launch Rectangle Pro will load a config file at ~/Library/Application Support/Rectangle Pro/RectangleProConfig.json if it is present and will rename that file with a time/date stamp so that it isn't read on subsequent launches. You can export a json file from the General tab of the Settings window.

I still do have plans to add something on par with the original ask, and I'll add your +1 to my backlog item for it.
